Helper/Requests.php

```diff
@@ -301,20 +301,23 @@ public function buildAddressData($billingAddress, $shippingAddress, $storeId, $r
     }
 
     /**
-     * @param array $request
      * @param $amount
      * @param $currencyCode
      * @param $reference
+     * @param $shopperConversionId
+     * @param array $request
      * @return array
      */
-    public function buildPaymentData($amount, $currencyCode, $reference, array $request = [])
+    public function buildPaymentData($amount, $currencyCode, $reference, $shopperConversionId,array $request = []): array
     {
+        var_dump($amount);
         $request['amount'] = [
             'currency' => $currencyCode,
             'value' => $this->adyenHelper->formatAmount($amount, $currencyCode)
         ];
 
         $request["reference"] = $reference;
+        $request["shopperConversionId"] = $shopperConversionId;
```

Comment:
If you decide to continue using this method instead of a separate data builder, this line should be covered with an if block checking the existence of `$shopperConversionId` as it will be an optional argument and could be `null` if it is not set explicitly.